### ISSUE TITLE
Provide a built-in collate_fn for DataLoader usage

### ISSUE THREAD

**Issue by fjktkm** [torchfont/torchfont#43]
## Summary

All four example scripts (`google_fonts.py`, `font_awesome.py`, `material_design_icons.py`, `source_han_sans.py`) define an identical `collate_fn` for padding variable-length glyph sequences in a `DataLoader`. This is a core use pattern that could be provided by the library.

## Duplicated code

Each example contains:

```python
def collate_fn(
    batch: Sequence[tuple[Tensor, Tensor, int, int]],
) -> tuple[Tensor, Tensor, Tensor, Tensor]:
    types_list = [types for types, _, _, _ in batch]
    coords_list = [coords for _, coords, _, _ in batch]
    style_label_list = [style for _, _, style, _ in batch]
    content_label_list = [content for _, _, _, content in batch]

    types_tensor = pad_sequence(types_list, batch_first=True, padding_value=0)
    coords_tensor = pad_sequence(coords_list, batch_first=True, padding_value=0.0)

    style_label_tensor = torch.as_tensor(style_label_list, dtype=torch.long)
    content_label_tensor = torch.as_tensor(content_label_list, dtype=torch.long)

    return types_tensor, coords_tensor, style_label_tensor, content_label_tensor
```

## Suggestion

Provide a built-in `collate_fn` (e.g. in `torchfont.utils` or `torchfont.transforms`) so users can write:

```python
from torchfont.utils import collate_fn

dataloader = DataLoader(dataset, batch_size=64, collate_fn=collate_fn)
```
